File: Index_Of_First_Occurrence/index_of_first_occurrence.py
def index_of_first_occurrence(haystack, needle):
    index = []
    if needle not in haystack:
        index.append(-1)
    else:
        for i in range(len(haystack)):
            if haystack[i:i+len(needle)] == needle:
                index.append(i)

    print(index[0])

index_of_first_occurrence("sadbutsad", "sad")
index_of_first_occurrence("leetcode", "leeto")

File: Index_Of_First_Occurrence/Index_Of_First_Occurrence.java
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static void main(String[] args) {
        indexOfFirstOccurrence("sadbutsad", "sad");
        indexOfFirstOccurrence("leetcode", "leeto");
    }

    public static void indexOfFirstOccurrence(String haystack, String needle) {
        List<Integer> index = new ArrayList<>();

        if (!haystack.contains(needle)){
            index.add(-1);
        }
        else{
            for (int i = 0; i <= haystack.length()- needle.length(); i++){
                if (haystack.substring(i, i + needle.length()).equals(needle)){
                    index.add(i);
                }
            }
        }

        System.out.println(index.get(0));
    }
}